LoopKitUI/Views/Settings Editors/BasalRateScheduleEditor.swift

```diff
@@ -91,9 +88,9 @@ public struct BasalRateScheduleEditor: View {
     
     private var savingMechanism: SavingMechanism<DailyQuantitySchedule<Double>> {
         switch mode {
-        case .modal:
+        case .settings, .legacySettings:
             return .asynchronous { quantitySchedule, completion in
-                self.syncSchedule(quantitySchedule.items) { result in
+                self.syncSchedule?(quantitySchedule.items) { result in
```

Comment:
Seems like no-oping here if `syncSchedule` is `nil` could lead to a subtle persistence bug if the client doesn't recognize that the argument is necessary under these modes. Consider adding an `assert` here against the optionality.